sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -632,6 +662,17 @@ pi_result piDevicesGet(pi_platform Platform, pi_device_type DeviceType,
     return PI_SUCCESS;
   }
 
+  // if devices are already captured in cache, return them from the cache.
```

Comment:
Why don't you populate the `Devices` on the first search loop at line 640?

Reply:
There are two different types of calls to piDevicesGet() from SYCL RT.
1. just get the number of devices
2. get all pi_devices for the platform
#1 is handled at line 640 and it does not pass the valid pointer for Devices param.
This code at 665 handles #2.

Reply:
Right. My point was that for #2 you could collect PI devices in the loop at 640 where you are just counting them now.

Reply:
Now, I changed the implementation to have individual pi_devices cache per platform.
So, I don't need to count the devices for each platform.